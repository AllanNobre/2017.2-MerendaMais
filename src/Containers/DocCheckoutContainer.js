import { connect } from 'react-redux';
import DocCheckoutScreen from '../screens/DocCheckoutScreen';
import { setDocReportPositive,
  setDocReportNegative,
  setDocObservation } from '../actions/reportActions';

const mapStateToProps = state => ({
  report: state.report.doc,
  observation: state.report.docObservation,
});

const mapDispatchToProps = dispatch => ({
  setDocReportPositive: key => dispatch(setDocReportPositive(key)),
  setDocReportNegative: key => dispatch(setDocReportNegative(key)),
  setDocObservation: observation => dispatch(setDocObservation(observation)),
});

const DocCheckoutContainer =
connect(mapStateToProps, mapDispatchToProps)(DocCheckoutScreen);

export default DocCheckoutContainer;
