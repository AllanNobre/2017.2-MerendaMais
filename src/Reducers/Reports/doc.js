const doc = [
  { label: 'Ausência de registros de imunização', markedYes: false, markedNo: false, status: false, key: 0 },
  { label: 'Ausência de registro de higienização dos utensílios', markedYes: false, markedNo: false, status: false, key: 1 },
  { label: 'Ausência de registro de capacitação ou treinamento de Merendeiras', markedYes: false, markedNo: false, status: false, key: 2 },
  { label: 'Ausência de registro de portabilidade da água', markedYes: false, markedNo: false, status: false, key: 3 },
  { label: 'Ausência de registro de exames periódicos dos manipuladores', markedYes: false, markedNo: false, status: false, key: 4 },
];

export default doc;
