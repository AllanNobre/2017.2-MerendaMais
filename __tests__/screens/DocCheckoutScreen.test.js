import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
// imported as a connected component!
import DocCheckoutContainer from '../../src/Containers/DocCheckoutContainer';
import doc from '../../src/Reducers/Reports/doc';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore();

const initialState = {
  report: {
    doc,
    docObservation: '',
  },
};

const store = mockStore(initialState);

describe('Testing StockFoodCheckoutScreen', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <DocCheckoutContainer />,
      { context: { store } },
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
