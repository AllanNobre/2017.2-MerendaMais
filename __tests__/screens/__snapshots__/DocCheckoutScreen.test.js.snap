// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DocCheckout Screen renders as expected 1`] = `
<ScrollViewMock
  style={
    Object {
      "backgroundColor": "white",
      "flex": 6,
      "flexDirection": "column",
      "marginBottom": 9,
    }
  }
>
  <Header
    backButton={true}
    subTitle="Documentação"
    title="Relatório"
  />
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontWeight": "bold",
            "paddingLeft": 10,
            "paddingTop": 5,
          }
        }
      >
        Sim
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontWeight": "bold",
            "paddingLeft": 20,
            "paddingTop": 5,
          }
        }
      >
        Nao
      </Text>
    </View>
    <View
      key="0"
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View>
        <Checkbox
          checkboxStyle={
            Object {
              "borderColor": "black",
              "borderRadius": 2,
              "borderWidth": 2,
              "flexWrap": "wrap",
              "height": 25,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 10,
              "width": 25,
            }
          }
          defaultColor="#fff"
          disabled={false}
          disabledColor={null}
          onSelect={[Function]}
          selected={false}
          selectedColor="#008000"
        />
      </View>
      <View>
        <Checkbox
          checkboxStyle={
            Object {
              "borderColor": "black",
              "borderRadius": 2,
              "borderWidth": 2,
              "flexWrap": "wrap",
              "height": 25,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 10,
              "width": 25,
            }
          }
          defaultColor="#fff"
          disabled={false}
          disabledColor={null}
          onSelect={[Function]}
          selected={false}
          selectedColor="#B22222"
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 1,
            "fontSize": 15,
            "paddingTop": 15,
          }
        }
      >
        Ausência de registros de imunização
      </Text>
    </View>
    <View
      key="1"
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View>
        <Checkbox
          checkboxStyle={
            Object {
              "borderColor": "black",
              "borderRadius": 2,
              "borderWidth": 2,
              "flexWrap": "wrap",
              "height": 25,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 10,
              "width": 25,
            }
          }
          defaultColor="#fff"
          disabled={false}
          disabledColor={null}
          onSelect={[Function]}
          selected={false}
          selectedColor="#008000"
        />
      </View>
      <View>
        <Checkbox
          checkboxStyle={
            Object {
              "borderColor": "black",
              "borderRadius": 2,
              "borderWidth": 2,
              "flexWrap": "wrap",
              "height": 25,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 10,
              "width": 25,
            }
          }
          defaultColor="#fff"
          disabled={false}
          disabledColor={null}
          onSelect={[Function]}
          selected={false}
          selectedColor="#B22222"
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 1,
            "fontSize": 15,
            "paddingTop": 15,
          }
        }
      >
        Ausência de registro de higienização dos utensílios
      </Text>
    </View>
    <View
      key="2"
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View>
        <Checkbox
          checkboxStyle={
            Object {
              "borderColor": "black",
              "borderRadius": 2,
              "borderWidth": 2,
              "flexWrap": "wrap",
              "height": 25,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 10,
              "width": 25,
            }
          }
          defaultColor="#fff"
          disabled={false}
          disabledColor={null}
          onSelect={[Function]}
          selected={false}
          selectedColor="#008000"
        />
      </View>
      <View>
        <Checkbox
          checkboxStyle={
            Object {
              "borderColor": "black",
              "borderRadius": 2,
              "borderWidth": 2,
              "flexWrap": "wrap",
              "height": 25,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 10,
              "width": 25,
            }
          }
          defaultColor="#fff"
          disabled={false}
          disabledColor={null}
          onSelect={[Function]}
          selected={false}
          selectedColor="#B22222"
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 1,
            "fontSize": 15,
            "paddingTop": 15,
          }
        }
      >
        Ausência de registro de capacitação ou treinamento de Merendeiras
      </Text>
    </View>
    <View
      key="3"
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View>
        <Checkbox
          checkboxStyle={
            Object {
              "borderColor": "black",
              "borderRadius": 2,
              "borderWidth": 2,
              "flexWrap": "wrap",
              "height": 25,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 10,
              "width": 25,
            }
          }
          defaultColor="#fff"
          disabled={false}
          disabledColor={null}
          onSelect={[Function]}
          selected={false}
          selectedColor="#008000"
        />
      </View>
      <View>
        <Checkbox
          checkboxStyle={
            Object {
              "borderColor": "black",
              "borderRadius": 2,
              "borderWidth": 2,
              "flexWrap": "wrap",
              "height": 25,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 10,
              "width": 25,
            }
          }
          defaultColor="#fff"
          disabled={false}
          disabledColor={null}
          onSelect={[Function]}
          selected={false}
          selectedColor="#B22222"
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 1,
            "fontSize": 15,
            "paddingTop": 15,
          }
        }
      >
        Ausência de registro de portabilidade da água
      </Text>
    </View>
    <View
      key="4"
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View>
        <Checkbox
          checkboxStyle={
            Object {
              "borderColor": "black",
              "borderRadius": 2,
              "borderWidth": 2,
              "flexWrap": "wrap",
              "height": 25,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 10,
              "width": 25,
            }
          }
          defaultColor="#fff"
          disabled={false}
          disabledColor={null}
          onSelect={[Function]}
          selected={false}
          selectedColor="#008000"
        />
      </View>
      <View>
        <Checkbox
          checkboxStyle={
            Object {
              "borderColor": "black",
              "borderRadius": 2,
              "borderWidth": 2,
              "flexWrap": "wrap",
              "height": 25,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 10,
              "width": 25,
            }
          }
          defaultColor="#fff"
          disabled={false}
          disabledColor={null}
          onSelect={[Function]}
          selected={false}
          selectedColor="#B22222"
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 1,
            "fontSize": 15,
            "paddingTop": 15,
          }
        }
      >
        Ausência de registro de exames periódicos dos manipuladores
      </Text>
    </View>
  </View>
  <View
    behavior="padding"
  >
    <View
      style={
        Object {
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 30,
        }
      }
    >
      <TextInput
        multiline={true}
        onChangeText={[Function]}
        placeholder="Observações (opcional)"
        style={
          Object {
            "borderRadius": 12,
            "borderWidth": 1,
            "fontSize": 37.5,
            "height": 333.5,
            "paddingLeft": 10,
            "textAlignVertical": "top",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <TouchableOpacity
    activeOpacity={0.2}
    key="setObservation"
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "#FF9500",
        "borderRadius": 7,
        "borderWidth": 1,
        "justifyContent": "flex-end",
        "marginBottom": 20,
        "marginHorizontal": 15,
        "marginTop": 30,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFF",
          "textAlign": "center",
        }
      }
    >
      Concluir
    </Text>
  </TouchableOpacity>
</ScrollViewMock>
`;
